app/desktop/studio_server/tool_servers_api.py

```diff
@@ -41,20 +47,34 @@ async def get_available_tool_servers(
     @app.get("/api/projects/{project_id}/tool_servers/{tool_server_id}")
     async def get_tool_server(
         project_id: str, tool_server_id: str
-    ) -> ExternalToolServer:
+    ) -> ExternalToolServerApiDescription:
         project = project_from_id(project_id)
-        tool = next(
+        tool_server = next(
             (
                 t
                 for t in project.external_tool_servers(readonly=True)
                 if t.id == tool_server_id
             ),
             None,
         )
-        if not tool:
-            raise HTTPException(status_code=404, detail="Tool not found")
+        if not tool_server:
+            raise HTTPException(status_code=404, detail="Tool Server not found")
 
-        return tool
+        # Get available tools based on server type
+        available_tools = []
+        if tool_server.type == ToolServerType.remote_mcp:
+            mcp_server = MCPServer(tool_server)
+            tools_result = await mcp_server.list_tools()
+            available_tools = tools_result.tools
+
+        return ExternalToolServerApiDescription(
+            name=tool_server.name,
+            type=tool_server.type,
+            description=tool_server.description,
+            properties=tool_server.properties,
+            parent=tool_server.parent,
```

Comment:
This is strange. Caller is calling with the parent ID to get this, they should already have the parent. An API returning the parent is generally backwards. Plus if we do want it for some reason, we can beat "parent" for a name. Use the actual type/name (project?). Just remove this.